### PR TITLE
HeatMap Implementation 

### PR DESCRIPTION
**Pull request by pseudo69**

### CODE REVIEW

File: Assets/CodeMonkey/Utils/Button_UI.cs [hunk 1]
```diff
@@ -131,10 +124,9 @@ public enum HoverBehaviour {
             if (MouseUpdate != null) MouseUpdate();
```
Comment (fmendez884) [inline Assets/CodeMonkey/Utils/Button_UI.cs:125]: remove white space

Comment (fmendez884) [reply Assets/CodeMonkey/Utils/Button_UI.cs:125]: line 125? OK maybe nvm actually

File: Assets/CodeMonkey/Utils/UI_Bar.cs [hunk 2]
```diff
@@ -39,50 +39,30 @@ public class Outline {
...
-
         public UI_Bar(Transform parent, Vector2 anchoredPosition, Vector2 size, Color backgroundColor, Color barColor, float sizeRatio, Outline outline) {
             SetupParent(parent, anchoredPosition, size);
             if (outline != null) SetupOutline(outline, size);
             SetupBackground(backgroundColor);
             SetupBar(barColor);
             SetSize(sizeRatio);
         }
-
         private void SetupParent(Transform parent, Vector2 anchoredPosition, Vector2 size) {
```
Comment (fmendez884) [inline Assets/CodeMonkey/Utils/UI_Bar.cs:66]: line 65, 77, 80, 85

change private functions to lower camel case??

File: Assets/CodeMonkey/Utils/UI_BarMultiple.cs [hunk 3]
```diff
@@ -51,7 +51,6 @@ public class Outline {
             barImageArr = barImageList.ToArray();
             SetSizes(defaultSizeList.ToArray());
         }
-
         private void SetupParent(Transform parent, Vector2 anchoredPosition, Vector2 size) {
```
Comment (fmendez884) [inline Assets/CodeMonkey/Utils/UI_BarMultiple.cs:54]: line 54, 64, 67

change private functions to lower camelcase

File: Assets/CodeMonkey/Utils/World_Bar.cs [hunk 4]
```diff
@@ -34,38 +33,25 @@ public class Outline {
...
         public World_Bar(Transform parent, Vector3 localPosition, Vector3 localScale, Color? backgroundColor, Color barColor, float sizeRatio, int sortingOrder, Outline outline = null) {
-            this.outline = outline;
             SetupParent(parent, localPosition);
             if (outline != null) SetupOutline(outline, localScale, sortingOrder - 1);
             if (backgroundColor != null) SetupBackground((Color)backgroundColor, localScale, sortingOrder);
             SetupBar(barColor, localScale, sortingOrder + 1);
             SetSize(sizeRatio);
         }
-
         private void SetupParent(Transform parent, Vector3 localPosition) {
```
Comment (fmendez884) [inline Assets/CodeMonkey/Utils/World_Bar.cs:43]: line 43,  49, 52, 55

change private functions to lower camelcase

File: Assets/CodeMonkey/Utils/World_Bar.cs [hunk 5]
```diff
@@ -74,60 +60,21 @@ public class Outline {
             bar.localScale = new Vector3(1,1,1);
             Transform barIn = UtilsClass.CreateWorldSprite(bar, "BarIn", Assets.i.s_White, new Vector3(localScale.x / 2f, 0), localScale, sortingOrder, barColor).transform;
```
Comment (fmendez884) [inline Assets/CodeMonkey/Utils/World_Bar.cs:61]: line 61 is the 2f in new Vector3 a magic number?

File: Assets/Grid.cs [hunk 6]
```diff
@@ -1,91 +1,160 @@
...
+
+            for (int x = 0; x < gridArray.GetLength(0); x++)
             {
-                debugTextArray[x, y] = UtilsClass.CreateWorldText(gridArray[x, y].ToString(), null, GetWorldPosition(x, y) + new Vector3(cellSize, cellSize) *.5f, 20, Color.white, TextAnchor.MiddleCenter);
-                Debug.DrawLine(GetWorldPosition(x, y), GetWorldPosition(x, y + 1), Color.white, 100f);
-                Debug.DrawLine(GetWorldPosition(x, y), GetWorldPosition(x + 1, y), Color.white, 100f);
+                for (int y = 0; y < gridArray.GetLength(1); y++)
+                {
+                    debugTextArray[x, y] = UtilsClass.CreateWorldText(gridArray[x, y].ToString(), null, GetWorldPosition(x, y) + new Vector3(cellSize, cellSize) * .5f, 30, Color.white, TextAnchor.MiddleCenter);
+                    Debug.DrawLine(GetWorldPosition(x, y), GetWorldPosition(x, y + 1), Color.white, 100f);
```
Comment (fmendez884) [inline Assets/Grid.cs:47]: Are there magic numbers in lines 46-56
46: new Vector3 - .5f, 30
47-56; 100f, x + 1

File: Assets/GridTester.cs [hunk 7]
```diff
@@ -6,23 +6,23 @@
...
-    private Grid grid;
+    [SerializeField] private HeatMapVisual heatMapVisual;
+
+    private BattleGrid grid;
+
     
    private void Start()
     {
-        grid = new Grid(4, 2, 10f, new Vector3(20,0));
+        grid = new BattleGrid(10, 10, 10f, Vector3.zero);
```
Comment (fmendez884) [inline Assets/GridTester.cs:16]: magic numbers?

File: Assets/HeatMapVisual.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+    public void SetGrid(BattleGrid grid)
+    {
+        this.grid = grid;
+        UpdateHeatMapVisual();
+
+        grid.OnGridValueChanged += Grid_OnGridValueChanged;
+    }
+
+    private void Grid_OnGridValueChanged(object sender, BattleGrid.OnGridValueChangedEventArgs e)
```
Comment (fmendez884) [inline Assets/HeatMapVisual.cs]: private function to be lower camel case

File: Assets/GridTester.cs (hunk 7 above)
Comment (pseudo69) [reply Assets/GridTester.cs:16]: All except one are not constants; these are values that we will be passing to instantiate a grid and it's size. Depending on the situation we might have larger or smaller grids. The exception is the size of the individual cell; we'll want to keep that consistent for art assets (importing tilesets, characters).

Comment (pseudo69) [reply Assets/GridTester.cs:16]: As we haven't decided on the size of the sprites or tilesets and this is a test class I haven't defined it as a const yet.

Comment (fmendez884) [reply Assets/GridTester.cs:16]: dont talk to me like that

File: Assets/Grid.cs [hunk 9]
```diff
@@ -1,91 +1,160 @@
...
     }
 
     public void SetValue(int x, int y, int value)
     {
-        if ( x >= 0 && y >=0 && x < width && y < height )
+        if (x >= 0 && y >= 0 && x < width && y < height)
         {
-          gridArray[x, y] = value;
-          debugTextArray[x, y].text = gridArray[x, y].ToString();
+            gridArray[x, y] = Mathf.Clamp(value, HEAT_MAP_MIN_VALUE, HEAT_MAP_MAX_VALUE);
```
Comment (fmendez884) [inline Assets/Grid.cs:90]: excess indentation line 90

**Review by fmendez884 — APPROVED** [review]

**Pull request merged**